### PR TITLE
[SC] Try to retrieve handle token address from child chain

### PR DESCRIPTION
**Pull request by hyunsooda**
## Proposed changes

- Moved runtime token register to the time when an initial token register is called via bridge setup phase.
- Removed invalid bridge pair comparison. This comparison becomes invalid when a node is rebooted.
- The current handle retrieve logic asks to parent chain to retrieve the corresponding handle token address. This PR changes to try to retrieve the handle token address from its own subbridge chain.

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [ ] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [ ] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [ ] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [ ] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: node/sc/api_bridge.go [hunk 1]
```diff
@@ -398,13 +406,6 @@ func (sb *SubBridgeAPI) DeregisterToken(cBridgeAddr, pBridgeAddr, cTokenAddr, pT
 		return ErrNoBridgeInfo
 	}
 
-	pTokenAddrCheck := cBi.GetCounterPartToken(cTokenAddr)
-	cTokenAddrCheck := pBi.GetCounterPartToken(pTokenAddr)
-
-	if pTokenAddr != pTokenAddrCheck || cTokenAddr != cTokenAddrCheck {
-		return errors.New("invalid toke pair")
-	}
-
```
Comment (hqjang-pepper) [inline node/sc/api_bridge.go:407]: I think removing this code is unnecessary. Can you explain why you removed this code?

File: node/sc/bridge_manager.go [hunk 2]
```diff
@@ -270,26 +271,41 @@ func (bi *BridgeInfo) UpdateInfo() error {
...
 }
 
-// handleRequestValueTransferEvent handles the given request value transfer event.
-func (bi *BridgeInfo) handleRequestValueTransferEvent(ev *RequestValueTransferEvent) error {
-	tokenType := ev.TokenType
-	tokenAddr := bi.GetCounterPartToken(ev.TokenAddress)
-	// TODO-Klaytn-Servicechain Add counterpart token address in requestValueTransferEvent
-	if tokenType != KLAY && tokenAddr == (common.Address{}) {
-		logger.Warn("Unregistered counter part token address.", "addr", tokenAddr.Hex())
-		ctTokenAddr, err := bi.counterpartBridge.RegisteredTokens(nil, ev.TokenAddress)
```
Comment (hqjang-pepper) [inline node/sc/bridge_manager.go:280]: Okay you removed the runtime token registering logic, and just return error when something goes wrong. But is it sure this error does not occur when we fix initial token register logic? I think we need to do some test for this.

File: node/sc/api_bridge.go (hunk 1 above)
Comment (hyunsooda) [reply node/sc/api_bridge.go:407]: The function `GetCounterPartToken()` lookup the `counterpartToken` property in `bridgeInfo` struct. The `counterpartToken` member is initialized when the function `RegisterToken()` is called. The `RegisterToken()` stands for recording the counterpart token address. If a node is rebooted, and no `RegisterToken()` is not called, `counterpartToken` becomes empty. Thus, the checker above is invalid.

File: node/sc/bridge_manager.go (hunk 2 above)
Comment (hyunsooda) [reply node/sc/bridge_manager.go:280]: The returns that return errors are actually fatal errors, meaning that the handle value transfer for the two types of the token must not be correctly handled. Please compare the previous code and this new change. You can find the key difference that a handle token address was retrieved from where. The handle token address was retrieved by asking for the counterpart chain's contract, not the own chain's contract. It opens a gateway to be victim's chain handles unintended value transfer events, although the unknown token contract has not been registered before.

**Review by hqjang-pepper — APPROVED** [review]

**Comment by hyunsooda** [comment]
An attacker can simply deploy the modified contract that emits the originally paired parent's contract by hardcoded address. A contract-based verification cannot be a solution to detect the integrity of emitted events.

**Pull request closed without merging**